crates/nu-cli/src/commands/classified/external.rs

```diff
@@ -110,7 +113,7 @@ pub(crate) async fn run_external_command(
         ));
     }
 
-    if command.has_it_argument() || command.has_nu_argument() {
```

Comment:
I don't think this is correct. I know we have a lot of code in `run_with_iterator_arg` that deals with filling in the nu variables, but having a nu variable isn't what should trigger `run_with_iterator_arg`.

This may potentially break some use case right now, but I'm planning on looping back and dealing with this later.